lib/ruby/stdlib/date/format.rb

```diff
@@ -577,7 +577,11 @@ def self._parse_ddd(str, e) # :nodoc:
   def self._parse(str, comp=true)
     # Newer MRI version (written in C converts non-strings to strings
     # and also has other checks like all ascii.
-    str = str.to_str if !str.kind_of?(::String) && str.respond_to?(:to_str)
+    if str.kind_of?(::String) || str.respond_to?(:to_str)
+      str = str.to_str if str.respond_to?(:to_str)
```

Comment:
the second check is for this case which seems very unlikely to me. Do you think it's worth to write a test for it?
```
class Sub < String
  undef_method :to_str
end
Date._parse Sub.new('20160505')
```

Reply:
not really - there wasn't a second check.